### ISSUE TITLE
Many buffered tests are failing on master

### ISSUE THREAD

**Issue by rolfv** [open-mpi/ompi#1065]
This problem is in master only and I tracked it down to this commit using git bisect.
```
[rvandevaart@drossetti-ivy5 ompi-master-rolfv]$ git bisect good
Bisecting: 0 revisions left to test after this (roughly 1 step)
[720fa860eee7df4f06eb5604058da57453e88b46]   changing the destruct function of list release API to release list items   caused a bug in oshmem application. Fixing the bug with this patch
[rvandevaart@drossetti-ivy5 ompi-master-rolfv]$ 
```
Although I cannot quite figure out what that commit is when I look at the log.  In any event, the errors can be seen on many different buffered send tests.
```
[rvandevaart@drossetti-ivy4 src]$ mpirun -np 2 MPI_Bsend_ator_c
--------------------------------------------------------------------------
MPITEST info  (0): Starting MPI_Bsend_ator: All Bsend TO Root test
MPITEST info  (0): Node spec MPITEST_comm_sizes[6]=2 too large, using 1
MPITEST info  (0): Node spec MPITEST_comm_sizes[22]=2 too large, using 1
MPITEST info  (0): Node spec MPITEST_comm_sizes[32]=2 too large, using 1
--------------------------------------------------------------------------
mpirun noticed that process rank 1 with PID 0 on node drossetti-ivy4 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

```
(gdb) where
#0  0x00007f16a17ef1d0 in ?? ()
#1  <signal handler called>
#2  0x00007f16a81e3abb in opal_memory_ptmalloc2_int_free (av=0x7f16a843f3a0, mem=0xb8e500)
    at ../../../../../opal/mca/memory/linux/malloc.c:4377
#3  0x00007f16a81e2317 in opal_memory_ptmalloc2_free (mem=0xb8e500) at ../../../../../opal/mca/memory/linux/malloc.c:3513
#4  0x00007f16a81e1051 in opal_memory_linux_free_hook (__ptr=0xb8e500, caller=0x7f16a8132687)
    at ../../../../../opal/mca/memory/linux/hooks.c:708
#5  0x00007f16a8132687 in opal_list_destruct (list=0xb8d250) at ../../opal/class/opal_list.c:108
#6  0x00007f16a4e6eeeb in opal_obj_run_destructors (object=0xb8d250) at ../../../../../opal/class/opal_object.h:460
#7  0x00007f16a4e6ff44 in mca_allocator_basic_finalize (base=0xb8d210)
    at ../../../../../opal/mca/allocator/basic/allocator_basic.c:362
#8  0x00007f16a8dd9fc4 in mca_pml_base_bsend_detach (addr=0x7fffa44e12c8, size=0x7fffa44e12c4)
    at ../../../../ompi/mca/pml/base/pml_base_bsend.c:190
#9  0x00007f16a8d52001 in PMPI_Buffer_detach (buffer=0x7fffa44e12c8, size=0x7fffa44e12c4) at pbuffer_detach.c:52
#10 0x0000000000402b76 in main (argc=1, argv=0x7fffa44e1668) at MPI_Bsend_ator_c.c:625
(gdb) 
```

**Comment by rolfv** [comment]
Looks like this was originally authored by @rppendya

**Comment by hppritcha** [comment]
Thanks for catching this.

**Comment by rhc54** [comment]
See https://github.com/open-mpi/ompi/pull/1066

**Comment by rolfv** [comment]
These errors are now gone so I will close this issue. I believe @rhc54 fixed this with #1066.
